support-frontend/assets/pages/[countryGroupId]/components/checkoutComponent.tsx

```diff
@@ -420,6 +423,15 @@ export function CheckoutComponent({
 
 	const formOnSubmit = async (formData: FormData) => {
 		setIsProcessingPayment(true);
+
+		/**  This validation has to happen on submit,
+		 *   as we cannot check it with form validation rules
+		 */
+		if (requireConfirmedEmail && email !== confirmedEmail) {
```

Comment:
What if `isEmailAddressReadOnly` is `true`, we won't  render the `confirm-email` input, so wouldn't it always fail this condition?

Reply:
great catch

Reply:
fixed in [9577d3f](https://github.com/guardian/support-frontend/pull/6742/commits/9577d3f9d9f2fb920b2d109042a11f6269b92872)